### PR TITLE
#253: Add Certificates in Secrets

### PR DESCRIPTION
**Pull request by DPigeon**
# Related Issue
- #253 

# Proposed changes
- Added certificates and keys into secrets
- CD uses the secrets to pass them into the angular dockerfile
- A script is adding them to the proper environment depending on the deployment

# Additional Info
- Since certificates were not hidden, a flow has been created for them to be hidden using secrets in our repo.

# Reviewer(s)
 - @wltrf 
 - @razine-bensari 
 - @abdulmansour

### CODE REVIEW

File: client/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # Client Build
 FROM trion/ng-cli as ng
-# Define the env config arg
+# Define args
 ARG ENVCONFIG
+ARG CERT
+ARG CERT_KEY
```
Comment (wltrf) [inline client/Dockerfile:6]: Should we really package the certificate and key inside the docker image? This exposes the private key and requires us to rebuild the image everytime we want to change the cert/key. I think it should simply be an env variable on the vm.

File: client/certs/money-tree.tech/cert.pem [hunk 2]
```diff
@@ -1,56 +0,0 @@
...
-Y3J5cHQub3JnMDwGA1UdHwQ1MDMwMaAvoC2GK2h0dHA6Ly9jcmwuaWRlbnRydXN0
-LmNvbS9EU1RST09UQ0FYM0NSTC5jcmwwHQYDVR0OBBYEFBQusxe3WFbLrlAJQOYf
-r52LFMLGMB0GA1UdJQQWMBQGCCsGAQUFBwMBBggrBgEFBQcDAjANBgkqhkiG9w0B
-AQsFAAOCAQEA2UzgyfWEiDcx27sT4rP8i2tiEmxYt0l+PAK3qB8oYevO4C5z70kH
-ejWEHx2taPDY/laBL21/WKZuNTYQHHPD5b1tXgHXbnL7KqC401dk5VvCadTQsvd8
-S8MXjohyc9z9/G2948kLjmE6Flh9dDYrVYA9x2O+hEPGOaEOa1eePynBgPayvUfL
-qjBstzLhWVQLGAkXXmNs+5ZnPBxzDJOLxhF2JIbeQAcH5H0tZrUlo5ZYyOqA7s9p
-O5b85o3AM/OJ+CktFBQtfvBhcJVd9wvlwPsk+uyOy2HI7mNxKKgsBTt375teA2Tw
-UdHkhVNcsAKX1H7GNNLOEADksd86wuoXvg==
------END CERTIFICATE-----
```
Comment (wltrf) [inline client/certs/money-tree.tech/cert.pem:56]: you could have kept the certificate here, this will be public anyway.

**Review by wltrf — CHANGES_REQUESTED** [review]

File: client/Dockerfile (hunk 1 above)
Comment (DPigeon) [reply client/Dockerfile:6]: Yeah you're right. What I am thinking right now is having the keys and certificates in their respective VM and then mounting a volume to the container when we start docker compose. I think this is a valid solution and all the scripting/secrets can be removed with that. As soon as we start docker compose, the client service will mount the certificates to the ./ssl directory on the container and nginx will be able to pick them up easily.

File: client/certs/money-tree.tech/cert.pem (hunk 2 above)
Comment (DPigeon) [reply client/certs/money-tree.tech/cert.pem:56]: I think I'll remove them anyway just for security reasons. Let's keep everything in the VM.

**Review by razine-bensari — APPROVED** [review]
Good work

**Review by wltrf — APPROVED** [review]

**Pull request merged**